RetailCoder.VBE/UI/Command/ReparseCommand.cs

```diff
@@ -26,6 +39,44 @@ protected override bool EvaluateCanExecute(object parameter)
 
         protected override void OnExecute(object parameter)
         {
+            if (_settings.CompileBeforeParse)
+            {
+                var failedNames = new List<string>();
+                using (var projects = _vbe.VBProjects)
+                {
+                    foreach (var project in projects)
+                    {
+                        using (project)
+                        {
+                            if (project.Protection != ProjectProtection.Unprotected)
+                            {
+                                continue;
+                            }
+
+                            if (!VBETypeLibsAPI.CompileProject(project))
+                            {
+                                failedNames.Add(project.Name);
+                            }
+                        }
+                    }
+                }
```

Comment:
I think it'd be nice to extract this into a separate method to keep `OnExecute` rather "high-level" Something like:

    If (_settings.CompileBeforeParse)
    {
        if (!AllProjectsCompile(out var failedNames)) 
        {
            // message box code
        }
    }